src/titiler/core/titiler/core/telemetry.py

```diff
@@ -0,0 +1,206 @@
+"""OpenTelemetry instrumentation for titiler.core."""
+
+import functools
+import inspect
+import sys
+from contextlib import contextmanager
+from typing import Any, Callable, Dict, Iterator, Optional, TypeVar, overload
+
+if sys.version_info >= (3, 10):
+    from typing import ParamSpec
+else:
+    from typing_extensions import ParamSpec
```

Comment:
I think typing extensions already has the `if sys.version_info` check, so while we make python>=3.10 we can just use `from typing_extensions import ParamSpec`